e2e.log

```diff
@@ -0,0 +1,1259 @@
+[13:15:36] Using gulpfile ~/amphtml/gulpfile.js
+[13:15:36] Starting 'e2e'...
+[13:15:36] Running [36mnpm install[39m to install packages in [36mbuild-system/tasks/e2e[39m...
+[13:15:39] [32mStarted[39m [36mAMP Dev Server[39m [32mat[39m [36mhttp://localhost:8000[39m
+[13:15:39] [32mServing[39m [36munminified[39m [32mJS[39m
+[13:15:39] Running tests...
+●●●●●●●●●●●○○●●●●●●●○●○●○●○●○●○●○●○●○●○●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●○●○○○●●●●○●●●●●●●●●○●●●●●●●○○●●○○○○○○●○○○●●●●●●●○○●●●●●●●●●○●●○●●●●●●●●●●●●○○○○●○○○○●○○○○●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●○○○○○○○○○○○○●○○●●●●●●●●●●●●●●●●●●●●●●●●●●○○●●●○●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●○○○●●●●●
+  1) amp-ad-exit
+       chrome
+          AMPHTML ads FIE environment 
+            
+             product1 clicked near the border:
+     JavascriptError: javascript error: cov_1xqb08d7d9 is not defined
+  (Session info: headless chrome=87.0.4280.88)
+      at Object.throwDecodedError (build-system/tasks/e2e/node_modules/selenium-webdriver/lib/error.js:550:15)
+      at parseHttpResponse (build-system/tasks/e2e/node_modules/selenium-webdriver/lib/http.js:565:13)
+      at Executor.execute (build-system/tasks/e2e/node_modules/selenium-webdriver/lib/http.js:491:26)
+      at runMicrotasks (<anonymous>)
+      at processTicksAndRejections (internal/process/task_queues.js:93:5)
+      at Driver.execute (build-system/tasks/e2e/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
+
+  2) amp-ad-exit
+       chrome
+          AMPHTML ads FIE environment 
+            
+             product1 ad opened:
+     JavascriptError: javascript error: cov_1xqb08d7d9 is not defined
```

Comment:
Searching for this error led me to https://github.com/istanbuljs/istanbuljs/issues/499, which mentions among other things the fact that a babel transform might be hoisting the coverage variable.

I wonder if it's worth playing with the order of babel plugins in `test-config.js`, `pre-closure-config.js`, and `unminified-config.js` by moving the instrumentation transform to the end.

Side note: I misspelled the plugin name as `instanbulPlugin` in `test-config.js` :)

Reply:
That sounds like a very plausible culprit. It's certainly possible to play with the order of plugins, and I did a lot of experimentation with varied results (though I do not recall every variation I attempted). TL;DR - Sometimes re-ordering results in coverage being generated for the binaries, rather than the source files. In theory it should be possible to do instrumentation at any step and transform the source maps along the way, but I never managed to get that working.